cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java

```diff
@@ -151,6 +153,13 @@ public ProgramController run(final Program program, ProgramOptions options) {
                                     BasicWorkflowToken.class);
     }
 
+    Map<String, String> workflowLocalDatasetNameMapping = null;
+    if (arguments.hasOption(ProgramOptionConstants.WORKFLOW_LOCAL_DATASET_NAME_MAPPING)) {
```

Comment:
Do we need this? Can't get it from the `DatasetFramework`? It's really get messy with this map flowing around with serializing and deserializing in different context. If not from the dataset framework, can it be in something more generic?